models/classes/media/LocalItemSource.php

```diff
@@ -314,4 +271,73 @@ protected function getItemDirectory()
         }
         return $this->itemService->getItemDirectory($this->item, $this->lang);
     }
+
+    private function searchDirectories(
+        string $parentLink,
+        array $acceptableMime,
+        int $depth,
+        int $childrenLimit
+    ): array {
+        if (!tao_helpers_File::securityCheck($parentLink)) {
+            throw new common_exception_Error(__('Your path contains error'));
+        }
+
+        $label = rtrim($parentLink, '/');
+        if (strrpos($parentLink, '/') !== false && substr($parentLink, -1) !== '/') {
+            $label = substr($parentLink, strrpos($parentLink, '/') + 1);
+            $parentLink = $parentLink . '/';
+        }
+
+        if (in_array($parentLink, ['', '/'])) {
+            $label = $this->getItem()->getLabel();
+            $parentLink = '/';
+        }
```

Comment:
Can we extract this to separate method?

Reply:
Yes, but there are already to many changes not directly related to the purpose of PR, as for code that needs to be backported at the 1.5 years old version, so I would avoid any additional refactoring.